collector/collector.go

```diff
@@ -0,0 +1,77 @@
+package collector
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/containous/traefik/cmd/traefik/anonymize"
+	"github.com/containous/traefik/configuration"
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/version"
+	"github.com/mitchellh/hashstructure"
+)
+
+const containousURL = "https://collect.traefik.io"
+
+// Collected data
+type data struct {
+	Version       string
+	Codename      string
+	BuildDate     string
+	Configuration string
+	Hash          string
+	EmitDate      string
+}
+
+// Collect anonymous data.
+func Collect(globalConfiguration *configuration.GlobalConfiguration) error {
+	anonConfig, err := anonymize.Do(globalConfiguration, false)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Stats sent on %s: %s", containousURL, anonConfig)
+
+	hashConf, err := hashstructure.Hash(globalConfiguration, nil)
+	if err != nil {
+		return err
+	}
+
+	data := &data{
+		Version:       version.Version,
+		Codename:      version.Codename,
+		BuildDate:     version.BuildDate,
+		Hash:          strconv.FormatUint(hashConf, 10),
+		Configuration: base64.StdEncoding.EncodeToString([]byte(anonConfig)),
+		EmitDate:      time.Now().Format(time.RFC3339),
+	}
+
+	buf := new(bytes.Buffer)
+	json.NewEncoder(buf).Encode(data)
+
+	_, err = makeHTTPClient().Post(containousURL, "application/json; charset=utf-8", buf)
```

Comment:
This might be a dumb question, but do we want this to retry/or log the data on failure? rather than just the error?

Reply:
the error is logged here: https://github.com/containous/traefik/pull/2447/files#diff-086f37ac2cd2ef1d326af9766ce3d7b6R282

Reply:
It's not critical for the application, we think the retry/log response is not needed in this case.